pkg/notebook/generate_test.go

```diff
@@ -0,0 +1,173 @@
+package notebook
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/tkestack/elastic-jupyter-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	JupyterNotebookName      = "jupyternotebook-sample"
+	JupyterNotebookNamespace = "default"
+	DefaultContainerName     = "notebook"
+	DefaultImage             = "busysandbox"
+	DefaultImageWithGateway  = "jupyter/base-notebook:python-3.8.6"
+	GatewayName              = "gateway"
+	GatewayNamespace         = "default"
+)
+
+var (
+	podSpec = v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Name:  DefaultContainerName,
+				Image: DefaultImage,
+			},
+		},
+	}
+
+	emptyNotebook = &v1alpha1.JupyterNotebook{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
We can remove the type meta here since the client will get the info via refletion.